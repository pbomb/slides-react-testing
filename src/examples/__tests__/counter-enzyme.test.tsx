import { render, mount } from 'enzyme';
import * as React from 'react';
import Counter from '../counter';

test('renders enabled buttons by default', () => {
  expect(render(<Counter />)).toMatchSnapshot();
});

test('renders disabled buttons', () => {
  expect(render(<Counter disabled={true} />)).toMatchSnapshot();
});

test('increments the counter when clicking the + button', () => {
  const wrapper = mount(<Counter />);
  wrapper.find('[data-testid="increment"]').simulate('click');
  expect(wrapper).toMatchSnapshot();
});

test('decrements the counter when clicking the - button', () => {
  const wrapper = mount(<Counter />);
  wrapper.find('[data-testid="decrement"]').simulate('click');
  expect(wrapper).toMatchSnapshot();
});
