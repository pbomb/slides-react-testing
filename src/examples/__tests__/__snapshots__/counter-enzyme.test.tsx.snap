// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`decrements the counter when clicking the - button 1`] = `
.css-0,
[data-css-0] {
  color: red;
}

<Counter>
  <div>
    <button
      data-testid="decrement"
      disabled={false}
      onClick={[Function]}
    >
      -
    </button>
    <Component
      clicks={-1}
    >
      <glamorous(div)
        clicks={-1}
        data-testid="count"
      >
        <div
          className="css-0"
          data-testid="count"
        >
          -1
        </div>
      </glamorous(div)>
    </Component>
    <button
      data-testid="increment"
      disabled={false}
      onClick={[Function]}
    >
      +
    </button>
  </div>
</Counter>
`;

exports[`increments the counter when clicking the + button 1`] = `
.css-0,
[data-css-0] {
  color: green;
}

<Counter>
  <div>
    <button
      data-testid="decrement"
      disabled={false}
      onClick={[Function]}
    >
      -
    </button>
    <Component
      clicks={1}
    >
      <glamorous(div)
        clicks={1}
        data-testid="count"
      >
        <div
          className="css-0"
          data-testid="count"
        >
          1
        </div>
      </glamorous(div)>
    </Component>
    <button
      data-testid="increment"
      disabled={false}
      onClick={[Function]}
    >
      +
    </button>
  </div>
</Counter>
`;

exports[`renders disabled buttons 1`] = `
.css-0,
[data-css-0] {
  color: black;
}

<div>
  <button
    data-testid="decrement"
    disabled=""
  >
    -
  </button>
  <div
    class="css-0"
    data-testid="count"
  >
    0
  </div>
  <button
    data-testid="increment"
    disabled=""
  >
    +
  </button>
</div>
`;

exports[`renders enabled buttons by default 1`] = `
.css-0,
[data-css-0] {
  color: black;
}

<div>
  <button
    data-testid="decrement"
  >
    -
  </button>
  <div
    class="css-0"
    data-testid="count"
  >
    0
  </div>
  <button
    data-testid="increment"
  >
    +
  </button>
</div>
`;
